src/typing/env_js.ml

```diff
@@ -29,6 +29,10 @@ let refinement_key names =
 let is_refinement name =
   (String.length name) >= 5 && (String.sub name 0 5) = "$REFI"
 
+let refined_object_from_key name =
+  let i = String.index name '.' in
+  String.sub name 6 (i - 6)
```

Comment:
Yeah, true on the dependency. You could always just wrap this stuff up in its own module here - I probably should've done that in the first place.